crates/egui/src/widget_text.rs

```diff
@@ -530,144 +540,133 @@ impl WidgetText {
-        }
+        self.map_rich_text(|text| text.strong())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn weak(self) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.weak()),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.weak())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn underline(self) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.underline()),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.underline())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn strikethrough(self) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.strikethrough()),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.strikethrough())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn italics(self) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.italics()),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.italics())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn small(self) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.small()),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.small())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn small_raised(self) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.small_raised()),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.small_raised())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn raised(self) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.raised()),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.raised())
     }
 
     /// Prefer using [`RichText`] directly!
+    #[inline]
     pub fn background_color(self, background_color: impl Into<Color32>) -> Self {
-        match self {
-            Self::RichText(text) => Self::RichText(text.background_color(background_color)),
-            Self::LayoutJob(_) | Self::Galley(_) => self,
-        }
+        self.map_rich_text(|text| text.background_color(background_color))
     }
 
     /// Returns a value rounded to [`emath::GUI_ROUNDING`].
     pub(crate) fn font_height(&self, fonts: &epaint::Fonts, style: &Style) -> f32 {
         match self {
+            Self::Text(_) => fonts.row_height(&FontSelection::Default.resolve(style)),
```

Comment:
This is the wrong `FontSelection` - we need to do something similar to https://github.com/emilk/egui/blob/b375687cbbba4c18077672b9497814ecccdfaebb/crates/egui/src/widget_text.rs#L406-L411